src/prometheus/ast.h

```diff
@@ -0,0 +1,126 @@
+#ifndef AST_H
+#define AST_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef enum {
+  PR_COUNTER,
+  PR_GAUGE,
+  PR_HISTOGRAM,
+  PR_SUMMARY,
+  PR_UNTYPED
+} pr_metric_type_t;
+
+typedef enum {
+  PR_METRIC_ENTRY,
+  PR_COMMENT_ENTRY,
+  PR_TYPE_ENTRY,
+  PR_HELP_ENTRY
+} pr_entry_type_t;
+
+typedef enum { PR_METRIC_FAMILY_ITEM, PR_COMMENT_ITEM } pr_item_type_t;
+
+typedef struct pr_label_s {
+  char *name;
+  char *value;
+  struct pr_label_s *next;
+} pr_label_t;
+
+typedef struct pr_timestamp_s {
+  bool has_value;
+  int64_t value;
+} pr_timestamp_t;
+
+typedef struct pr_metric_entry_s {
+  char *name;
+  pr_label_t *labels;
+  double value;
+  pr_timestamp_t *timestamp;
+} pr_metric_entry_t;
+
+typedef struct pr_comment_entry_s {
+  char *text;
+} pr_comment_entry_t;
+
+typedef struct pr_type_entry_s {
+  char *name;
+  pr_metric_type_t tp;
+} pr_type_entry_t;
+
+typedef struct pr_help_entry_s {
+  char *name;
+  char *hint;
+} pr_help_entry_t;
+
+typedef struct pr_entry_s {
+  pr_entry_type_t tp;
+  union {
+    pr_metric_entry_t *metric;
+    pr_comment_entry_t *comment;
+    pr_type_entry_t *type;
+    pr_help_entry_t *help;
+  } body;
+} pr_entry_t;
+
+typedef struct pr_item_s pr_item_t;
+
+typedef struct pr_metric_s pr_metric_t;
+
+typedef struct pr_metric_s {
+  pr_label_t *labels;
+  double value;
+  pr_timestamp_t *timestamp;
+  pr_metric_t *next;
+} pr_metric_t;
+
+typedef struct pr_metric_family_s {
+  char *name;
+  char *help;
+  pr_metric_type_t tp;
+  pr_metric_t *metric_list;
+} pr_metric_family_t;
+
+typedef pr_comment_entry_t pr_comment_t;
+
+typedef struct pr_item_s {
+  pr_item_type_t tp;
+  union {
+    pr_metric_family_t *metric_family;
+    pr_comment_t *comment;
```

Comment:
Why this struct is special?  It could use `pr_comment_entry_t` like the other structs do:
```suggestion
typedef struct pr_item_s {
  pr_item_type_t tp;
  union {
    pr_metric_family_t *metric_family;
    pr_comment_entry_t *comment;
```